code/modules/mob/living/simple_mob/subtypes/animal/_animal_harness.dm

```diff
@@ -0,0 +1,169 @@
+/obj/item/storage/internal/animal_harness
+	abstract_type = /obj/item/storage/internal/animal_harness
+	name = "animal harness"
+	color = COLOR_BEASTY_BROWN
+	max_w_class = ITEMSIZE_LARGE
+	max_storage_space = INVENTORY_STANDARD_SPACE
+
+	/// Null, or a list of (/obj/item/foo = "item key"), or (/path = ("item key", proc/handler)).
+	var/list/attachable_types

```

Comment:
I'd prefer to have a more flexible concept of mob inventory in general, and converting to that might be a future project. As is, though, it would be pretty squirrely to do this in another way.